src/main/scala/cromwell/engine/io/gcs/GoogleCloudStorage.scala

```diff
@@ -23,8 +23,26 @@ import scala.language.postfixOps
 import scala.util.{Failure, Success, Try}
 
 object GoogleCloudStorage {
-  def apply(appName: String, credential: Credential, jsonFactory: JsonFactory, httpTransport: HttpTransport): GoogleCloudStorage = {
-    GoogleCloudStorage(new Storage.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build())
+
+  val RefreshTokenOptionKey = "refresh_token"
+
+  lazy val cromwellAuthenticated: Try[GoogleCloudStorage] = {
+    for {
+      cromwellCredentials <- Try(GoogleCredentialFactory.fromCromwellAuthScheme)
+      conf <- GoogleConfiguration.gcloudConf
+    } yield apply(conf.appName, cromwellCredentials)
+  }
+
+  def userAuthenticated(workflowOptions: WorkflowOptions): Option[GoogleCloudStorage] = {
+    for {
+      conf <- GoogleConfiguration.gcloudConf.toOption
```

Comment:
We can make it something like this instead
```
def userAuthenticated(workflowOptions: WorkflowOptions): Option[GoogleCloudStorage] = {
    GoogleConfiguration.gcloudConf map { conf =>
      for {
        token <- workflowOptions.get(RefreshTokenOptionKey).toOption
        userCredentials <- GoogleCredentialFactory.fromUserAuthScheme(token)
      } yield apply(conf.appName, userCredentials)
    } get
  }
```
This way we'll fail if we can't find a google configuration (which we should anyway, with the error message being the validation error), and we still get an `Option[Credential]` depending on if the user configuration (clientId / Secret + refresh token) was provided.